superset/db_engine_specs.py

```diff
@@ -973,12 +974,15 @@ def fetch_data(cls, cursor, limit):
     @staticmethod
     def create_table_from_csv(form, table):
         """Uploads a csv file and creates a superset datasource in Hive."""
-        def get_column_names(filepath):
-            with open(filepath, 'rb') as f:
-                return next(unicodecsv.reader(f, encoding='utf-8-sig'))
-
-        table_name = form.name.data
-        filename = form.csv_file.data.filename
+        def convert_to_hive_type(col_type):
+            """maps tableschema's types to hive types"""
+            if col_type == "number":
```

Comment:
I think having a dict for the mapping would be cleaner also would allow for people to easily add more types if need be.

Reply:
Agree. Added it to config